integration-tests/deployment/keystone/deploy.go

```diff
@@ -0,0 +1,537 @@
+package keystone
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/smartcontractkit/chainlink/integration-tests/deployment"
+	v1 "github.com/smartcontractkit/chainlink/integration-tests/deployment/jd/node/v1"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/durationpb"
+
+	capabilitiespb "github.com/smartcontractkit/chainlink-common/pkg/capabilities/pb"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
+	kcr "github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+type DeployRequest struct {
+	RegistryChain uint64
+	Menv          deployment.MultiDonEnvironment
+
+	DonToCapabilities map[string][]kcr.CapabilitiesRegistryCapability                   // from external source
+	NodeIDToNop       map[string]capabilities_registry.CapabilitiesRegistryNodeOperator // maybe should be derivable from JD interface but doesn't seem to be notion of NOP in JD
+}
+
+type DeployResponse struct {
+	Changeset *deployment.ChangesetOutput
+	DonToId   map[string]uint32
+}
+
+func Deploy(ctx context.Context, lggr logger.Logger, req DeployRequest) (*DeployResponse, error) {
```

Comment:
this is the primary function to drive the deployment and configure the register.

i want to come back and simplify the code into bit size pieces, but it works as is with multiple tests and need to move forward https://smartcontract-it.atlassian.net/browse/KS-456